runtime/src/hardened_unpack.rs

```diff
@@ -167,24 +166,46 @@ pub fn unpack_snapshot<A: Read, P: AsRef<Path>>(
     )
 }
 
-fn is_valid_snapshot_archive_entry(parts: &[&str], kind: tar::EntryType) -> bool {
-    let like_storage = Regex::new(r"^\d+\.\d+$").unwrap();
-    let like_slot = Regex::new(r"^\d+$").unwrap();
+fn all_digits(v: &str) -> bool {
+    for x in v.chars() {
+        if !x.is_numeric() {
+            return false;
+        }
+    }
+    true
+}
 
-    trace!("validating: {:?} {:?}", parts, kind);
+fn like_storage(v: &str) -> bool {
+    let mut periods = 0;
+    let mut saw_numbers = false;
+    for x in v.chars() {
+        if !x.is_numeric() {
+            if x == '.' {
+                if periods > 0 || !saw_numbers {
+                    return false;
+                }
+                saw_numbers = false;
+                periods += 1;
+            } else {
+                return false;
+            }
+        } else {
+            saw_numbers = true;
+        }
+    }
+    saw_numbers
```

Comment:
maybe this needs to be changed `periods == 1 && saw_numbers`?

I think "232323" incorrectly returns false.

Reply:
yes, fixed and added test